akka-docs/rst/java/stream/stream-composition.rst

```diff
@@ -247,7 +247,7 @@ The propagation of the individual materialized values from the enclosed modules
 |
 
 To implement the above, first, we create a composite :class:`Source`, where the enclosed :class:`Source` have a
-materialized type of :class:`Promise<BoxedUnit>`. By using the combiner function ``Keep.left()``, the resulting materialized
+materialized type of :class:`CompletableFuture<Optional<Integer>>>`. By using the combiner function ``Keep.left()``, the resulting materialized
```

Comment:
[This](http://doc.akka.io/docs/akka/2.4/java/stream/stream-quickstart.html#Transforming_and_consuming_simple_streams) refers to [CompositionDocTest.java](https://github.com/akka/akka/blob/4a7cd147313404eb8137e56163e54f06c23640d0/akka-docs/rst/java/code/docs/stream/CompositionDocTest.java#L247)

where the code is as follows:

```
    final Source<Integer, CompletableFuture<Optional<Integer>>> source = Source.<Integer>maybe(); 
```
So `CompletableFuture<Optional<Integer>>` is correct.